mycroft/audio/audioservice.py

```diff
@@ -82,16 +83,41 @@ def get_services(services_folder):
     return sorted(services, key=lambda p: p.get('name'))
 
 
-def load_services(config, bus, path=None):
+def setup_service(service_module, config, bus):
+    """Run the appropriate setup function and return created service objects.
+
+    Arguments:
+        service_module: Python module to run
+        config (dict): Mycroft configuration dict
+        bus (MessageBusClient): Messagebus interface
+    Returns:
+        (list) List of created services.
     """
-        Search though the service directory and load any services.
+    if (hasattr(service_module, 'autodetect') and
+            callable(service_module.autodetect)):
+        try:
+            return service_module.autodetect(config, bus)
+        except Exception as e:
+            LOG.error('Failed to autodetect. ' + repr(e))
```

Comment:
I was about to change this but when going through the code I realized these were intentionally set to error. This is since these are expected to fail to load if the prerequisite isn't installed so we don't want to clog up the log with traces for modules that aren't installed.

I'd like to wait to change this until third party optional modules are removed from core and all services are expected to load.

Reply:
I added a comment to the relevant issue for extracting optional services so we don't forget about this.